# Question 3) What is the correlation between the number of surviors and fare with respect to same Age_Group and Station?

In [1]:
# Descriptive labelling Step
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
titanic_df = pd.read_csv('Titanic.csv')

titanic_df['Survival'] = titanic_df.Survived.map({0 : 'Died', 1 : 'Survived'})
titanic_df['Class'] = titanic_df.Pclass.map({1 : 'First Class', 2 : 'Second Class', 3 : 'Third Class'})

In [3]:
titanic_df = titanic_df.drop(titanic_df.columns[[0,3,6,7,8,10]], axis=1)
titanic_df = titanic_df.dropna()
titanic_df['Survival'] = titanic_df.Survived.map({0 : 'Died', 1 : 'Survived'})
titanic_df['Class'] = titanic_df.Pclass.map({1 : 'First Class', 2 : 'Second Class', 3 : 'Third Class'})
titanic_df['Station'] = titanic_df.Embarked.map({'C' : 'Cherbourg', 'Q' : 'Queenstown', 'S' : 'Southampton'})
titanic_df['Age_Group'] = pd.cut(titanic_df.Age, [0, 10, 19, 29, 59, 80], labels = ['Child', 'Teenager', 'Young', 'Adult', 'Oldsters']).astype(object)

In [4]:
Survived_table = titanic_df.pivot_table(values='Survived', index=['Age_Group', 'Station'], aggfunc=np.mean)
Survived_table

Survived
Age_Group Station              
Adult     Cherbourg    0.568966
          Queenstown   0.181818
          Southampton  0.388889
Child     Cherbourg    0.888889
          Queenstown   0.000000
          Southampton  0.588235
Oldsters  Cherbourg    0.400000
          Queenstown   0.000000
          Southampton  0.222222
Teenager  Cherbourg    0.650000
          Queenstown   0.666667
          Southampton  0.324324
Young     Cherbourg    0.605263
          Queenstown   0.400000
          Southampton  0.293785

In [7]:
Fare_table = titanic_df.pivot_table(values='Fare', index=['Age_Group', 'Station'], aggfunc=np.mean)
Fare_table

Fare
Age_Group Station               
Adult     Cherbourg    88.813076
          Queenstown   26.893182
          Southampton  29.483030
Child     Cherbourg    21.031022
          Queenstown   29.125000
          Southampton  32.196567
Oldsters  Cherbourg    60.117520
          Queenstown    7.750000
          Southampton  40.782172
Teenager  Cherbourg    56.885845
          Queenstown    7.481950
          Southampton  28.161147
Young     Cherbourg    55.258884
          Queenstown    7.745000
          Southampton  21.823749

In [8]:
Fare_table['Fare'].corr(Survived_table['Survived'])

0.25044095427656438

I have used pivot_table function of dataframe in order to segregate the data in table format. I have ceated two tables namely Survived_table and Fare_table. I found only 0.2504 corelation between surviver and fare.